### PR TITLE
Fixes cmake build of OGDI driver - issue #4817

### PR DESCRIPTION
**Pull request by andrew-aitchison**
Fixes cmake build of OGDI driver - issue #4817
We now use pkgconfig to find tirpc headers.

## What are related issues/pull requests?
This does not fix the DODS driver since from issue #4817:
DODS support will be dropped in GDAL 3.5, so you can leave that out and build with cmake .. -DGDAL_USE_DAP=OFF to disable it.

## Tasklist

 - [ ] Review
 - [ ] Adjust for comments
 - [ ] All CI builds and checks have passed

## Environment

* OS: Ubuntu 21.10
* Compiler: g++ v11.2

### CODE REVIEW

File: cmake/modules/packages/FindOGDI.cmake [hunk 1]
```diff
@@ -12,28 +12,50 @@
...
-find_path(OGDI_INCLUDE_DIR ecs.h PATH_SUFFIXES ogdi)
+find_package(PkgConfig QUIET)
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_OGDI QUIET OGDI)
+  set(OGDI_VERSION_STRING ${PC_OGDI_VERSION})
+  set(OGDI_INCLUDE_DIRS ${PC_OGDI_INCLUDE_DIRS})
+endif()
+
+find_path(OGDI_INCLUDE_DIR
+          NAMES ecs.h rpc/rpc.h
```
Comment (rouault) [inline cmake/modules/packages/FindOGDI.cmake:23]: do we need to look for rpc/rpc.h ? This is an implementation detail. GDAL just includes ecs.h

File: cmake/modules/packages/FindOGDI.cmake [hunk 2]
```diff
@@ -12,28 +12,50 @@
...
+          HINTS ${PC_OGDI_INCLUDE_DIRS})
```
Comment (rouault) [inline cmake/modules/packages/FindOGDI.cmake:24]: shouldn't the PATH_SUFFIXES ogdi should be kept (for non-pkgconfig cases)

File: cmake/modules/packages/FindOGDI.cmake (hunk 1 above)
Comment (andrew-aitchison) [reply cmake/modules/packages/FindOGDI.cmake:23]: Possibly. The original error was

[100%] Building CXX object ogr/ogrsf_frmts/ogdi/CMakeFiles/ogr_OGDI.dir/ogrogdidatasource.cpp.o
In file included from /home/werdna/git/gdal/ogr/ogrsf_frmts/ogdi/ogrogdidatasource.cpp:30:
In file included from /home/werdna/git/gdal/ogr/ogrsf_frmts/ogdi/ogrogdi.h:40:
/usr/include/ogdi/ecs.h:9:10: fatal error: 'rpc/rpc.h' file not found
#include <rpc/rpc.h>
         ^~~~~~~~~~~
1 error generated.

The actual change in Ubuntu is that there are now two transport-independent RPC libraries  tirpc and ntirpc, with headers moved from /usr/include/rpc/ to /usr/include/tirpc/rpc/ and /usr/include/ntirpc/rpc/
It is not clear to me that without pkgconfig, cmake would find rpc/rpc.h

How do I generate a non-pkgconfig, cmake build ?

File: cmake/modules/packages/FindOGDI.cmake (hunk 2 above)
Comment (andrew-aitchison) [reply cmake/modules/packages/FindOGDI.cmake:24]: Sounds reasonable. I'll test it when I test the non-pkgconfig, cmake build.

File: cmake/modules/packages/FindOGDI.cmake (hunk 1 above)
Comment (rouault) [reply cmake/modules/packages/FindOGDI.cmake:23]: > It is not clear to me that without pkgconfig, cmake would find rpc/rpc.h

indeed, but I suspect that OGDI_INCLUDE_DIR only includes the directory of the first file found, here ecs.h ? (genuinely asking, as I don't have the target env. immediately available)

> How do I generate a non-pkgconfig, cmake build ?

I would just rename /usr/share/pkgconfig/ogdi.pc to ogdi.pc.disabled. But is there a ogdi.pc file at all ??

**Comment by rouault** [comment]
why closing it ?

**Comment by andrew-aitchison** [comment]
Sorry.
Trying to resync my local and github repos.
New pull request coming soon.

**Comment by rouault** [comment]
> Trying to resync my local and github repos. New pull request coming soon.

ok, no problem. I'd just note that  git rebase + git push -f   or git merge can be used to update an existing pull request too.

**Pull request closed without merging**